src/components/FeatureTrainingModal.tsx

```diff
@@ -0,0 +1,234 @@
+import type {VideoReadyForDisplayEvent} from 'expo-av';
+import React, {useCallback, useEffect, useState} from 'react';
+import {View} from 'react-native';
+import {GestureHandlerRootView} from 'react-native-gesture-handler';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useOnboardingLayout from '@hooks/useOnboardingLayout';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import Navigation from '@libs/Navigation/Navigation';
+import variables from '@styles/variables';
+import * as User from '@userActions/User';
+import CONST from '@src/CONST';
+import Button from './Button';
+import CheckboxWithLabel from './CheckboxWithLabel';
+import Lottie from './Lottie';
+import LottieAnimations from './LottieAnimations';
+import type DotLottieAnimation from './LottieAnimations/types';
+import Modal from './Modal';
+import SafeAreaConsumer from './SafeAreaConsumer';
+import Text from './Text';
+import VideoPlayer from './VideoPlayer';
+
+// Aspect ratio and height of the video.
+// Useful before video loads to reserve space.
+const VIDEO_ASPECT_RATIO = 1280 / 960;
+
+const MODAL_PADDING = variables.spacing2;
+
+type VideoLoadedEventType = {
+    srcElement: {
+        videoWidth: number;
+        videoHeight: number;
+    };
+};
+
+type VideoStatus = 'video' | 'animation';
+
+type FeatureTrainingModalProps = {
+    /** Animation to show when video is unavailable */
+    animation?: DotLottieAnimation;
+
+    /** URL for the video */
+    videoURL: string;
+
+    /** Title for the modal */
+    title?: string;
+
+    /** Describe what is showing */
+    description?: string;
+
+    /** Whether to show `Don't show me this again` option */
+    shouldShowDismissModalOption?: boolean;
+
+    /** Text to show on primary button */
+    confirmText: string;
+
+    /** A callback to call when user confirms the tutorial */
+    onConfirm?: () => void;
+
+    /** Text to show on secondary button */
+    helpText?: string;
+
+    /** Link to navigate to when user wants to learn more */
+    onHelp?: () => void;
+};
+
+function FeatureTrainingModal({
```

Comment:
iOS - Not found page when clicking `Got it`

https://github.com/Expensify/App/assets/29673073/1761159d-4866-445f-b626-f0d84f6c19b5